### ISSUE TITLE
Support recording traces from the browser

### ISSUE THREAD

**Issue by OrKoN** [w3c/webdriver-bidi#745]
Chrome allows recording trace information from the browser https://pptr.dev/api/puppeteer.tracing.start

Does it make sense (and possible) to have this functionality cross-browser?

**Comment by whimboo** [comment]
@OrKoN do you have more details about the data that is actually included in those traces?

**Comment by OrKoN** [comment]
I would suggest leaving it to be browser-specific, e.g., as a standard way to get browser-specific performance and debugging metrics with opaque output.

**Comment by jgraham** [comment]
I agree that there is user benefit in being able to access tracing via BiDi, but if the output format is non-standard I'm not sure how useful it is to have a standard command to start/stop recording vs having browser specific commands, which have browser-specific options. If we had at least a common output format to target there would be more obvious benefit in having a common API to support producing that data format.

**Comment by taladar** [comment]
Maybe parts of the OpenTelemetry trace spec could be used as a standard format? Possibly with some standardized span names but individual browsers could also add their own (especially for lower layers in the span hierarchy)?

**Comment by OrKoN** [comment]
Note that traces are quite large and it might not be feasible to put into a base64 blob. In CDP, the tracing data is sent periodically using the event or a blog stream. Puppeteer relies on streams. The workflow is as following:

1) Tracing.start(a bunch of options on what to trace and transferMode set 'ReturnAsStream').
2) Tracing.stop (returns streamId).
3) IO domain https://chromedevtools.github.io/devtools-protocol/tot/IO/ is used to fetch the data.
